From c90601c09ab8bc3a2cbf0798c5ac4a8d132ec5f4 Mon Sep 17 00:00:00 2001
From: Izhar Ameer Shaikh <izhar.ameer.shaikh@xilinx.com>
Date: Wed, 9 Sep 2020 23:46:21 -0700
Subject: [PATCH 1582/1851] firmware: xilinx: Add IOCTL support for AIE ISR
 Clear

commit dad81715d7d3caa362a46eb7b05226d8470300d9 from
https://github.com/Xilinx/linux-xlnx.git

Latching of AIE NPI Interrupts is present in Versal ES1 Silicon Rev,
however it has been removed from ES2 rev.
As a result on ES1, in order to use the interrupt, a client needs to
request PMC to clear/ack the interrupt.

Provide an EEMI IOCTL to serve the same purpose. Note that, this will
only be applicable for ES1 rev. For ES2 and other non-silicon platforms,
this call will essentially be a NOP in the firmware.

Signed-off-by: Izhar Ameer Shaikh <izhar.ameer.shaikh@xilinx.com>
Reviewed-by: Wendy Liang <wendy.liang@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/firmware/xilinx/zynqmp.c     | 1 +
 include/linux/firmware/xlnx-zynqmp.h | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/drivers/firmware/xilinx/zynqmp.c b/drivers/firmware/xilinx/zynqmp.c
index 159ec5e9e927..c1d3320e5114 100644
--- a/drivers/firmware/xilinx/zynqmp.c
+++ b/drivers/firmware/xilinx/zynqmp.c
@@ -531,6 +531,7 @@ static inline int versal_is_valid_ioctl(u32 ioctl_id)
 	case IOCTL_USB_SET_STATE:
 	case IOCTL_OSPI_MUX_SELECT:
 	case IOCTL_GET_LAST_RESET_REASON:
+	case IOCTL_AIE_ISR_CLEAR:
 		return 1;
 	default:
 		return 0;
diff --git a/include/linux/firmware/xlnx-zynqmp.h b/include/linux/firmware/xlnx-zynqmp.h
index 0b5f88c97953..f8f00a75b11d 100644
--- a/include/linux/firmware/xlnx-zynqmp.h
+++ b/include/linux/firmware/xlnx-zynqmp.h
@@ -182,6 +182,8 @@ enum pm_ioctl_id {
 	IOCTL_USB_SET_STATE,
 	/* IOCTL to get last reset reason */
 	IOCTL_GET_LAST_RESET_REASON,
+	/* AIE ISR Clear */
+	IOCTL_AIE_ISR_CLEAR,
 };
 
 enum pm_query_id {
-- 
2.31.1

